colorizer_data/bin/tfe_open.py

```diff
@@ -113,8 +128,16 @@ def main():
 
     args = parser.parse_args()
     dataset_path = os.path.abspath(args.dataset_path)
-    tfe_port = args.tfe_port
-    directory_port = args.port
+
+    tfe_port = get_available_port(args.tfe_port)
+    directory_port = get_available_port(args.port)
```

Comment:
super nit but curious, what's the chance these two calls end up returning the same port? Looks like get_available_port() binds a socket, gets a port and then immediately releases it ( if `with` block exists), which makes that port available again for the next call. In you demo this worked great, so this approach is totally fine for the use case here.